terraform/dms/dms.tf

```diff
@@ -86,8 +86,14 @@ resource "aws_dms_replication_task" "default" {
 
   migration_type = each.value.task.migration_type
 
-  replication_task_settings = local.task_settings
-  table_mappings            = local.table_mappings
+  replication_task_settings = jsonencode(
+      merge(
+        jsondecode(local.default_task_settings),
+        coalesce(each.value.task.settings_overrides, {})
+      )
+    )
+
+  table_mappings = jsonencode(coalesce(each.value.task.table_mappings, jsondecode(local.default_table_mappings)))
```

Comment:
Why do we have a difference in the way replication_task_settings and table mappings are calculated? 

Shouldn't table_mappings use a merge as well? Something I'm missing maybe.

Reply:
It's because `merge` doesn't do a recursive merge, so I think for table mappings it should be a matter of define everything for a config, or use the default

https://github.com/alphagov/paas-cf/pull/3103/commits/8f17f4596307218bf45b09ac820f6752c2456bb3

Reply:
I've used `merge` in the task settings because the task settings are much more expansive, and there's a greater likelihood of wanting to change only a small part